src/client/app/components/BarChartComponent.tsx

```diff
@@ -0,0 +1,257 @@
+						if (barDuration.asDays() != 1) {
+							// subtracting one extra day caused by day ending at midnight of the next day.
+							// Going from DB unit timestamp that is UTC so force UTC with moment, as usual.
+							timeRange += ` - ${moment.utc(barReading.endTimestamp).subtract(1, 'days').format('ll')}`;
+						}
+						hoverText.push(`<b> ${timeRange} </b> <br> ${label}: ${readingValue.toPrecision(6)} ${unitLabel}`);
+					});
+
+					// This variable contains all the elements (x and y values, bar chart, etc.) assigned to the data parameter of the Plotly object
+					datasets.push({
+						name: label,
+						x: xData,
+						y: yData,
+						text: hoverText,
+						hoverinfo: 'text',
+						type: 'bar',
+						marker: { color: getGraphColor(colorID, DataType.Group) }
+					});
+				}
+			}
+		}
+	}
+
+	const handleRelayout = (e: PlotRelayoutEvent) => {
+		// This event emits an object that contains values indicating changes in the user's graph, such as zooming.
+		// These values indicate when the user has zoomed or made other changes to the graph.
+		if (e['xaxis.range[0]'] && e['xaxis.range[0]']) {
+			// The event signals changes in the user's interaction with the graph.
+			// this will automatically trigger a refetch due to updating a query arg.
+			const startTS = moment.utc(e['xaxis.range[0]'])
+			const endTS = moment.utc(e['xaxis.range[1]'])
+			const workingTimeInterval = new TimeInterval(startTS, endTS);
+			dispatch(graphSlice.actions.updateTimeInterval(workingTimeInterval));
+		}
+	}
+
+	// Assign all the parameters required to create the Plotly object (data, layout, config) to the variable props, returned by mapStateToProps
+	// The Plotly toolbar is displayed if displayModeBar is set to true (not for bar charts)
+
+	if (raw) {
+		return <h1><b>${translate('bar.raw')}</b></h1>
+	}
+	let enoughData = false;
+	datasets.forEach(dataset => {
+		if (dataset.x.length > 1) {
+			enoughData = true
+		}
+	})
+	if (datasets.length === 0) {
+		return <h1>
+			{`${translate('select.meter.group')}`}
+		</h1>
+	} else if (!enoughData) {
+		// This normal so plot.
+		return <h1>
+			{`${translate('threeD.no.data')}`}
+		</h1>
+	} else {
+		return (
+			<Plot
+				data={datasets as Plotly.Data[]}
+				style={{ width: '100%', height: '100%' }}
+				onRelayout={handleRelayout}
+				useResizeHandler={true}
+				config={{
+					displayModeBar: false,
+					responsive: true
+				}}
+				layout={{
+					barmode: (barStacking ? 'stack' : 'group'),
+					bargap: 0.2, // Gap between different times of readings
+					bargroupgap: 0.1, // Gap between different meter's readings under the same timestamp
+					showlegend: true,
+					legend: { x: 0, y: 1.1, orientation: 'h' },
+					yaxis: {
+						title: unitLabel, showgrid: true,
+						gridcolor: '#ddd', fixedrange: true
+					},
+					xaxis: {
+						rangeslider: { visible: true },
+						// rangeselector: { visible: true },
```

Comment:
Needed?

Reply:
Not sure. Let me know. I'm still testing things & potential ui controls. I removed the redraw and restore, which i think should come back in some way. Similarly, I added a range slider to bar ,but not entirely sure if its here to stay. We need to solidify the UI design In the near future.

Reply:
Fixed For now.